homeassistant/components/epson/config_flow.py

```diff
@@ -25,18 +25,31 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
 
     async def async_step_import(self, import_config):
         """Import a config entry from configuration.yaml."""
-        return await self.async_step_user(import_config)
+        for entry in self._async_current_entries(include_ignore=True):
+            if import_config[CONF_HOST] == entry.data[CONF_HOST]:
+                return self.async_abort(reason="already_configured")
+        await self.async_set_unique_id()
+        self._abort_if_unique_id_configured()
+        return self.async_create_entry(
+            title=import_config.pop(CONF_NAME), data=import_config
+        )
 
     async def async_step_user(self, user_input=None):
         """Handle the initial step."""
         errors = {}
         if user_input is not None:
             try:
-                await validate_projector(
-                    self.hass, user_input[CONF_HOST], user_input[CONF_PORT]
-                )
+                projector = await validate_projector(self.hass, user_input[CONF_HOST])
+                serial_no = await projector.get_serial_number()
+                await self.async_set_unique_id(serial_no)
```

Comment:
Please only wrap the line(s) that can raise in the try... except block. Move the other lines down to the `else:` block.